spec/jobs/irs_attempts_events_batch_job_spec.rb

```diff
@@ -52,25 +52,27 @@
         redis_client = IrsAttemptsApi::RedisClient.new
         events.each do |event|
           redis_client.write_event(
-            event_key: event[:event_key], jwe: event[:jwe],
-            timestamp: event[:timestamp]
+            event_key: event[:event_key],
+            jwe: event[:jwe],
+            timestamp: event[:timestamp],
           )
         end
       end
 
       it 'batches and writes attempt events to an encrypted file' do
-        # puts "PUBLIC KEY = "
-        # puts Base64.strict_encode64(public_key.to_der)
-
-        # expect(IrsAttemptsApi::EnvelopeEncryptor).to receive(:encrypt).with(
-        #  data: events.collect{|e| e[:jwe]}.join("\r\n"),
-        #  timestamp: start_time,
-        #  public_key: OpenSSL::PKey::RSA.new(encoded_public_key),
-        # )
+        expect(IrsAttemptsApi::EnvelopeEncryptor).to receive(:encrypt).with(
+          data: events.pluck(:jwe).join("\r\n"),
+          timestamp: start_time,
+          public_key_str: encoded_public_key,
+        )
```

Comment:
Now that we aren't having a nightmare about object equality with instances of `PKey::RSA`, we can assert that we pass the expected parameters into `.encrypt`.